### PR TITLE
Force windows_service to restart in order to cope with restart error:

### PR DESCRIPTION
**Pull request by stefanwb**
"Cannot stop service 'Datadog Agent (DatadogAgent)' because it has dependent services.'"

Signed-off-by: Stefan Wessels Beljaars <swesselsbeljaars@schubergphilis.com>

### CODE REVIEW

File: providers/monitor.rb [hunk 1]
```diff
@@ -43,6 +43,7 @@ def whyrun_supported?
   service 'datadog-agent' do
     service_name node['datadog']['agent_name']
     provider service_provider unless service_provider.nil?
+    restart_command "powershell restart-service #{node['datadog']['agent_name']} -Force" if node['os'] == 'windows'
```
Comment (olivielpeau) [inline providers/monitor.rb]: for consistency with the rest of the cookbook, could you use `node['platform_family'] == 'windows'` as the condition here?

Comment (stefanwb) [reply providers/monitor.rb]: Updated in amended commit.

**Comment by stefanwb** [comment]
Last commit amended, added `restart_command` to last mentioned service now as well

**Comment by stefanwb** [comment]
Apparently our internal version bump came along and caused a merge conflict. This has been fixed.
@olivielpeau Could you have another look at this PR?

**Comment by stefanwb** [comment]
I'v added -Force to the stop_command.
Also I've squashed the branch back to 3 commits, our in-house version bumps got included by accident. Should be clean enough to merge now.

**Comment by olivielpeau** [comment]
Thanks @stefanwb! The code changes look good to me, I'll double-check they work well on the platforms we support (sadly Windows isn't part of the CI yet) and merge.

**Pull request merged**